src/plugins/newsfeed/public/lib/driver.test.ts

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { NewsfeedApiDriver } from './driver';
+import { storageMock } from './storage.mock';
+
+const kibanaVersion = '8.0.0';
+const userLanguage = 'en';
+const fetchInterval = 2000;
+
+describe('NewsfeedApiDriver', () => {
+  let driver: NewsfeedApiDriver;
+  let storage: ReturnType<typeof storageMock.create>;
+
+  beforeEach(() => {
+    storage = storageMock.create();
+    driver = new NewsfeedApiDriver(kibanaVersion, userLanguage, fetchInterval, storage);
+  });
+
+  describe('shouldFetch', () => {
+    it('returns true if no value is present in the storage', () => {
+      storage.getLastFetchTime.mockReturnValue(undefined);
+      expect(driver.shouldFetch()).toBe(true);
+      expect(storage.getLastFetchTime).toHaveBeenCalledTimes(1);
+    });
+
+    it('returns true if last fetch time precedes page load time', () => {
+      storage.getLastFetchTime.mockReturnValue(new Date(322642800000)); // 1980-03-23
+      expect(driver.shouldFetch()).toBe(true);
+    });
+
+    it('returns false if last fetch time is recent enough', () => {
+      storage.getLastFetchTime.mockReturnValue(new Date(3005017200000)); // 2065-03-23
```

Comment:
Are we creating an issue for 43 years in the future? Should we calculate this date to ensure it's always in the future?